unittests/security_group_tests.cpp

```diff
@@ -0,0 +1,396 @@
+ (data (i32.const 768) "get_active_security_group output buffer must match the input")
+)
+)=====";
+
+std::vector<char> participants_payload( participants_t names ) {
+   fc::datastream<std::vector<char>> ds;
+   fc::raw::pack(ds, names);
+   return ds.storage();
+}
+
+BOOST_AUTO_TEST_CASE(test_security_group_intrinsic) {
+
+   eosio::testing::tester chain1;
+   using namespace eosio::chain::literals;
+
+   chain1.create_accounts( {"alice"_n,"bob"_n,"charlie"_n} );
+   chain1.produce_blocks(3);
+
+   chain1.create_accounts({ "addmember"_n, "rmmember"_n, "ingroup"_n, "getgroup"_n });
+   chain1.produce_block();
+
+   chain1.set_code( "addmember"_n, propose_security_group_participants_add_wast );
+   chain1.set_code( "rmmember"_n, propose_security_group_participants_remove_wast );
+   chain1.set_code( "ingroup"_n, assert_in_security_group_wast );
+   chain1.set_code( "getgroup"_n, assert_get_security_group_wast );
+   chain1.produce_block();
+   
+   chain1.push_action( "eosio"_n, "setpriv"_n, "eosio"_n, fc::mutable_variant_object()("account", "addmember"_n)("is_priv", 1));
+   chain1.push_action( "eosio"_n, "setpriv"_n, "eosio"_n, fc::mutable_variant_object()("account", "rmmember"_n)("is_priv", 1));
+   chain1.produce_blocks(24);
+
+   chain1.set_producers( {"alice"_n,"bob"_n} );
+   chain1.produce_blocks(3); // Starts new blocks which promotes the proposed schedule to pending
+   BOOST_REQUIRE_EQUAL( chain1.control->active_producers().version, 1u );
+
+   BOOST_TEST_REQUIRE(chain1.push_action(  eosio::chain::action({}, "addmember"_n, {}, participants_payload({"alice"_n, "bob"_n})), "addmember"_n.to_uint64_t() ) == "");
+   chain1.produce_blocks(11);
+   BOOST_CHECK_EQUAL(chain1.control->proposed_security_group_participants().size() , 2);
+   chain1.produce_blocks(12);
+   BOOST_CHECK_EQUAL(chain1.control->proposed_security_group_participants().size() , 0);
+   BOOST_CHECK(chain1.control->in_active_security_group(participants_t({"alice"_n, "bob"_n})));
+
+   BOOST_TEST_REQUIRE(chain1.push_action(  eosio::chain::action({}, "rmmember"_n, {}, participants_payload({"alice"_n})), "rmmember"_n.to_uint64_t() ) == "");
+   BOOST_CHECK_EQUAL(chain1.control->proposed_security_group_participants().size() , 1);
+   chain1.produce_blocks(11+12);
+   BOOST_CHECK(!chain1.control->in_active_security_group(participants_t({"alice"_n})));
+
+   BOOST_TEST_REQUIRE(chain1.push_action(  eosio::chain::action({}, "ingroup"_n, {}, participants_payload({"bob"_n})), "ingroup"_n.to_uint64_t() ) == "");
+
+   eosio::chain::security_group_info_t grp{
+      .version = 2,
+      .participants = {"bob"_n}
+   };
+
+   fc::datastream<std::vector<char>> strm;
+   fc::raw::pack(strm, grp);
+
+   BOOST_TEST_REQUIRE(chain1.push_action(eosio::chain::action({}, "getgroup"_n, {}, strm.storage()), "getgroup"_n.to_uint64_t()) == "");
+   chain1.produce_blocks(11);
+   chain1.control->abort_block();
+
+   /// Test snapshot recovery
```

Comment:
Add a step to add to the proposed security group so that you can also verify below that the global_property_object is also recovered